File: src/main/java/src/com/study/design/pattern/solid/single/responsibility/Persistence.java
package src.com.study.design.pattern.solid.single.responsibility;

import java.io.PrintStream;

/**
 * Handles the responsibility of persisting objects
 */
class Persistence {
    void saveJournal(Journal journal) throws Exception {

        String filename = "MyJournal";

        try (PrintStream out = new PrintStream(filename)) {
            out.println(journal.toString());
        }
    }
}

File: src/main/kotlin/src/com/study/design/pattern/solid/single/responsibility/Persistence.kt
package src.com.study.design.pattern.solid.single.responsibility

import java.io.PrintStream

/**
 * Handles the responsibility of persisting objects
 */
class Persistence {

    fun saveJournal(journal: Journal) = PrintStream(FILE_NAME).use { out -> out.println(journal.toString()) }

    companion object{
        const val FILE_NAME = "MyJournal"
    }
}